src/script/page/Modals/InviteModal/InviteModal.tsx

```diff
@@ -0,0 +1,122 @@
+ *
+ */
+
+import React, {useState} from 'react';
+import {createRoot, Root} from 'react-dom/client';
+import {Runtime} from '@wireapp/commons';
+
+import {t} from 'Util/LocalizerUtil';
+
+import ModalComponent from 'Components/ModalComponent';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import Icon from 'Components/Icon';
+import {Config} from '../../../Config';
+import {UserState} from '../../../user/UserState';
+
+export interface InviteModalProps {
+  readonly userState: UserState;
+  onClose?: () => void;
+}
+
+const {BRAND_NAME: brandName} = Config.getConfig();
+
+const InviteModalComponent: React.FC<InviteModalProps> = ({userState, onClose}) => {
+  const [isInviteMessageSelected, setIsInviteMessageSelected] = useState<boolean>(false);
+  const {self: selfUser} = useKoSubscribableChildren(userState, ['self']);
+  const userName = selfUser.username();
+  const inviteMessage = userName
+    ? t('inviteMessage', {brandName: brandName, username: `@${userName}`})
+    : t('inviteMessageNoEmail', brandName);
+
+  const metaKey = Runtime.isMacOS() ? t('inviteMetaKeyMac') : t('inviteMetaKeyPc');
+  const inviteHint = isInviteMessageSelected ? t('inviteHintSelected', metaKey) : t('inviteHintUnselected', metaKey);
+
+  const onTextClick = () => setIsInviteMessageSelected(true);
+  const onBlur = () => setIsInviteMessageSelected(false);
+
+  const onClick = (ev: React.MouseEvent<HTMLTextAreaElement, MouseEvent>) => {
+    (ev.target as HTMLTextAreaElement).select();
+    onTextClick();
+  };
+
+  const onFocus = (ev: React.FocusEvent<HTMLTextAreaElement, Element>) => {
+    ev.target.select();
+    onTextClick();
+  };
+
+  return (
+    <div className="invite-modal">
+      <ModalComponent isShown onBgClick={onClose} onClosed={onClose} data-uie-name="modal-invite">
+        <div className="modal__header">
+          <h2 className="modal__header__title" data-uie-name="status-modal-title">
+            {t('inviteHeadline', brandName)}
+          </h2>
+
+          <button type="button" className="modal__header__button" onClick={onClose} data-uie-name="do-close">
+            <Icon.Close />
+          </button>
+        </div>
+
+        <div className="modal__body invite-modal__body">
+          <textarea
+            defaultValue={inviteMessage}
+            onClick={onClick}
+            onFocus={onFocus}
+            onBlur={onBlur}
+            className="modal__input reset-textarea invite-modal__message"
+            dir="auto"
+          />
+
+          <div className="modal__info invite-modal__info" data-bind="text: inviteHint">
+            {inviteHint}
+          </div>
+        </div>
+      </ModalComponent>
+    </div>
+  );
+};
+
+export default InviteModalComponent;
+
+let modalContainer: HTMLDivElement;
+let reactRoot: Root;
+
+const cleanUp = () => {
+  if (modalContainer) {
+    reactRoot.unmount();
+    document.body.removeChild(modalContainer);
```

Comment:
I think appending to document body will break some styles as it did in my PR can you check that and if the styles are broken do it to #wire-main instead like here: https://github.com/wireapp/wire-webapp/pull/13230/files#diff-95ed7641b4ddf02e5e622ac2d81f29df88d8a604c8b8226fdc5ee828a61db8d1R104-R115